biscuit-auth/src/token/builder.rs

```diff
@@ -757,15 +770,42 @@ impl From<biscuit_parser::builder::Scope> for Scope {
         match scope {
             biscuit_parser::builder::Scope::Authority => Scope::Authority,
             biscuit_parser::builder::Scope::Previous => Scope::Previous,
-            biscuit_parser::builder::Scope::PublicKey(pk) => {
-                Scope::PublicKey(PublicKey::from_bytes(&pk).expect("invalid public key"))
-            }
+            biscuit_parser::builder::Scope::PublicKey(pk) => Scope::PublicKey(
+                PublicKey::from_bytes(
+                    &pk.key,
+                    match pk.algorithm {
+                        biscuit_parser::builder::Algorithm::Ed25519 => Algorithm::Ed25519,
```

Comment:
maybe a pair From<> instance for converting between builder algorithm and datalog algorithm would be nicer

Reply:
done in bdb8c74